src/librustc/hir/mod.rs

```diff
@@ -46,9 +47,17 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn hir(self) -> Hir<'tcx> {
         Hir { tcx: self, map: &self.hir_map }
     }
+
+    pub fn hir_id_parent_module(self, id: HirId) -> DefId {
+        self.parent_module(DefId::local(id.owner))
+    }
 }
 
 pub fn provide(providers: &mut Providers<'_>) {
+    providers.parent_module = |tcx, id| {
+        let hir = tcx.hir();
+        hir.get_module_parent(hir.as_local_hir_id(id).unwrap())
```

Comment:
Can we make this function `get_module_parent` and `get_module_parent_node` less public? The query is always the better choice, right?